### DISCUSSION TITLE
WiFi Components inside Circuit Boxes cannot link to chat or communicate with Components outside Circuit Boxes, and vice versa.

### DISCUSSION THREAD

**Discussion by Sarynkia** [FakeFishGames/Barotrauma#14465] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Disclaimer for the disclaimer: Only texture replacement mods were in use.

It appears that WiFi Components placed inside Circuit Boxes cannot link to chat (no output) even with "link to chat" enabled in the server settings. Placing the exact same WiFi Component (or combinations thereof) outside Circuit Boxes however works correctly.

This occurred in MP while I'm cleaning up the Sub myself, I have not tried replicating this issue in SP or doing so with other players in the server.

### Reproduction steps

1. Start a Multiplayer campaign.
2. Place a "link to chat"-enabled WiFi Component inside a Circuit Box.
3. (Connect the WiFi Component to a Signal Out pin in the Circuit Box, and the pin to a text display if desired to verify)
4. Send whatever message to the WiFi Component. Verify that no messages in chat appear in the corresponding channel.
5. Take the WiFi Component _out of_ the Circuit Box for comparison.
6. Send whatever message to it again.
7. Profit. Messages should appear normally in chat.

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.5.9.1 (Summer Update Hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by Sarynkia** [comment]
EDIT:
Made further testing in MP with just myself.

It appears that signals received by WiFi Components _inside_ Circuit Boxes cannot be directly transmitted to WiFi components that are _outside_ circuit boxes and vice versa. Link to chat also counts.

**Steps to Reproduce:**
1. Start up an MP campaign.
2. Put down a Terminal and a Circuit Box. Connect the Signal Out pin of the Terminal to a Signal In pin of the Circuit Box
3. Place a WiFi Component inside the Circuit Box. Set an arbitrary channel for it (suppose, 301).
4. Place another WiFi Component outside the Circuit Box, set the same channel for it (in our case, 301).
5. Place a Text Display and link it to the WiFi Component in Step 4.
6. Send whatever message to the Circuit Box.
7. Verify that no text is displayed/no text changes are observed, nor are there messages sent to in-game chat on channel 301.
8. Profit.

If further comparison is desired, here are the additional steps to verify that "Box to Box" WiFi communication works:
9. Place a second WiFi Component inside the Circuit Box (or a separate Box, both work) with the same channel as set in Step 3 (in our case, again, 301).
10. Wire this WiFi Component's Signal Out pin to the Circuit Box's Signal Out pin.
11. Connect the Circuit Box's Signal Out pin to a text display.
12. Verify that text is displayed normally.
13. Profit.

**Comment by Regalis11** [comment]
Addresed in https://github.com/FakeFishGames/Barotrauma-development/pull/5653 (PR in our private repo)

**Comment by Jasontti** [comment]
Tested a fix for this.
Closing discussion as resolved.
